### PR TITLE
Add `contribute/framework` doc

### PR DESCRIPTION
**Pull request by kylegach**
## What I did

Add a page at the `/docs/<framework>/contribute/framework` route explaining how to contribute a Storybook framework

### CODE REVIEW

**Review by shilman — APPROVED** [review]
This is great!

Historically, people have wanted to contribute frameworks directly to the monorepo--I wonder if this doc should somehow preempt that.

I also wonder about a step on distributing the framework, e.g. tweet and mention `@storybookjs`. Or if the pattern described here will automatically get picked up by the integrations catalog, mention that?

File: docs/contribute/framework.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+The library or libraries your framework supports may have different major versions available. Carefully consider which versions of each library your framework will support. You will need to account for the changes within those different versions or split your framework into different versions/packages itself to support each library version. To help ongoing maintenance, please consider adding integration tests for the various library versions your framework supports.
+
+### 3. Write the README
+
+Before writing any code, we advise you to write up a helpful README. It should, at minimum, contain installation instructions and a list of available features. Use the [README for `@storybook/nextjs`](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/README.md) as a template. By writing the documentation first, it becomes a guide for your other work.
```
Comment (jonniebigodes) [inline docs/contribute/framework.md:23]: @kylegach we can probably rename this  "Write documentation". What do you think?

File: docs/contribute/framework.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Test it in a fresh project using a Storybook set up as close as possible to your framework. For example, for `@storybook/nextjs`, which uses React and Webpack5, start with a project that uses `@storybook/react` and `@storybook/builder-webpack5`. Follow the installation instructions from your README and ensure everything works as expected. Remember to test the various versions, configs, and options for the libraries you’re supporting.
+
+### 6. Let us know!
+
+Once it's fully tested and released, please let us know about your framework by either announcing it in the `#showcase` channel of the [Storybook Discord](https://discord.gg/storybook) or tweeting it and mentioning `@storybookjs`. It's our hope that well-made community frameworks can eventually move into the Storybook codebase and be considered "official" support.
+
+## Thank you
+
+We deeply appreciate your help in making Storybook work for more developers and the libraries in which they build. If you have any suggestions to improve this document, please use the “Edit” link below.
```
Comment (jonniebigodes) [inline docs/contribute/framework.md:145]: @kylegach, if this is going to be added to the documentation, we don't need this. Instead, we could replace this with something linking to the other parts of the documentation of how to contribute, something like:
```md
---
Learn about other ways you can contribute to Storybook:
- [Documentation](./documentation.md) 
- [Examples](./new-snippets.md) 
- [Code](./code.md)
```

If you're ok with this, we can adjust the other contribution pages

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@kylegach, this looks great. Just left a couple of items for you to look at. Let me know if you're ok with it, and we'll go from there also while we're at it. It seems that you left out the toc.js entry so we can hook this up to the documentation. Was this intentional or not?

**Review by jonniebigodes — APPROVED** [review]
Thanks for taking care of the feedback so promptly @kylegach! Appreciate it. All good on my end! Merge when able

**Pull request merged**